king_phisher/client/widget/managers.py

```diff
@@ -373,3 +374,61 @@ def signal_activate_popup_menu_copy(self, menuitem, column_ids):
 
 	def signal_activate_popup_menu_delete(self, menuitem):
 		self._call_cb_delete()
+
+class TimeSelector(gui_utilities.GladeGObject):
+	dependencies = gui_utilities.GladeDependencies(
+		children=(
+			'spinbutton_hour',
+			'spinbutton_minute'
+		),
+		top_level=(
+			'ClockHourAdjustment',
+			'ClockMinuteAdjustment'
+		)
+	)
+	top_gobject = 'popover'
+
+class TimeSelectorButtonManager(object):
+	def __init__(self, button, application, value=None):
+		"""
+		:param button: The button to activate TimeSelectorPopover()
+		:type button: :py:class:`Gtk.ToggleButton`
+		:param application: The application instance which this object belongs to.
+		:param value: The present datetime value (initialized to 00:00)
+		:type datetime.time: The value, (initialized to NoneType) will always return datetime.time(hr, min)
+		"""
+		self.popover = TimeSelector(application)
+		self.button = button
+		self.application = application
+		self._hour_spin = self.popover.gobjects['spinbutton_hour']
+		self._minute_spin = self.popover.gobjects['spinbutton_minute']
+		self.time = value or datetime.time(0, 0)
+		self.button.connect('toggled', self.signal_button_toggled)
+		self.popover.popover.set_relative_to(self.button)
+		self.button.set_label(f"{0:02}:{0:02}")
+
+	def __repr__(self):
+		return "<{0} time='{1:%H:%M}' >".format(self.__class__.__name__, self.time)
+
+	def signal_button_toggled(self, _):
+		if not self.button.get_active():
```

Comment:
flip the logic save some space

Reply:
I tried this initially, and it caused a weird loop where it would always be stuck open. I don't know why that occurred, but this was how I resolved it.